File: santander-dev-bootcamp/src/main/java/me/dio/santanderdevbootcamp/domain/model/user/Address.java
package me.dio.santanderdevbootcamp.domain.model;

import jakarta.persistence.Embeddable;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Embeddable
@Getter
@Setter
public class Address {
    @NotNull
    private String CEP;
    @NotNull
    private String Rua;
    @NotNull
    private String Bairro;
    @NotNull
    private char[] Estado=new char[2];
}

File: kotlinRestApi/src/main/kotlin/me/dio/kotlinRestApi/system/entity/Address.kt
package me.dio.kotlinRestApi.system.entity

import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
class Address(
    @Column(nullable = false)
    var zipCode: String = "",
    @Column(nullable = false)
    var street: String = "",
    @Column(nullable = true)
    var number: Int? = null,
)